connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -564,6 +570,9 @@ public Schema asConnectSchema(JsonNode jsonSchema) {
      * @return JsonNode-encoded version
      */
     private JsonNode convertToJsonWithEnvelope(Schema schema, Object value) {
```

Comment:
I wonder if it is worth changing the name of this method to make it clear that `null` can be returned. Perhaps something like `toEnvelopeOrNull`?

Reply:
The fact is that in both cases if the `schema` and `value` are `null` the return is `null`. You can follow that in `convertToJson`.
To make it simpler I moved the logic to `fromConnectData`

Reply:
Which "sort of" duplicates some logic, not sure if it's ok this way, please let me know what you think